doozer/doozerlib/cli/fbc.py

```diff
@@ -425,6 +470,175 @@ async def run(self):
+        :return: Existing FBC build record if found, None otherwise
+        """
+        fbc_name = KonfluxFbcRebaser.get_fbc_name(operator_meta.distgit_key)
+        logger = self._logger.getChild(f"[{operator_meta.distgit_key}]")
+
+        logger.info("Checking for existing FBC build...")
+        where = {
+            "name": fbc_name,
+            "group": self.runtime.group,
+            "outcome": str(KonfluxBuildOutcome.SUCCESS),
+        }
+
+        # Search for FBC builds that contain this bundle build NVR
+        existing_fbc_build = await anext(
+            self._fbc_db.search_builds_by_fields(
+                where=where, array_contains={"bundle_nvrs": bundle_build.nvr}, limit=1
+            ),
+            None,
+        )
+        if existing_fbc_build:
+            logger.info(f"Found existing FBC build: {existing_fbc_build.nvr}")
+            return existing_fbc_build
+        return None
+
+    async def run(self):
+        """Rebase and build fbc fragments for given operator NVRs or all latest operator NVRs for the group and assembly"""
+        self._logger.info("Starting FBC rebase and build process...")
+
+        runtime = self.runtime
+        runtime.initialize(config_only=True)
+        assembly = runtime.assembly
+        if assembly is None:
+            raise ValueError("Assemblies feature is disabled for this group. This is no longer supported.")
+        assert runtime.konflux_db is not None, "konflux_db is not initialized. Doozer bug?"
+        konflux_db = runtime.konflux_db
+        konflux_db.bind(KonfluxBuildRecord)
+
+        # Get operator builds and corresponding bundle builds
+        dgk_operator_builds = await FbcRebaseCli.get_operator_builds(runtime, self.operator_nvrs)
+        dgk_bundle_builds = await FbcRebaseCli.get_bundle_builds(
+            self._db_for_bundles, runtime.group, list(dgk_operator_builds.values()), strict=False
+        )
+        not_found = [dgk for dgk, bundle_build in dgk_bundle_builds.items() if bundle_build is None]
+        if not_found:
+            raise IOError(f"Bundle builds not found for {not_found}. Please build the bundles first.")
+
+        assert runtime.source_resolver is not None, "source_resolver is not initialized. Doozer bug?"
+        assert runtime.group_config is not None, "group_config is not initialized. Doozer bug?"
+
+        self._logger.info(f"Processing {len(dgk_operator_builds)} operator(s): {list(dgk_operator_builds.keys())}")
+
+        # Check for existing FBC builds if force is not set
+        if not self.force:
+            existing_builds = []
+            for dgk, bundle_build in dgk_bundle_builds.items():
+                operator_meta = runtime.image_map[dgk]
+                existing_fbc_build = await self._check_existing_fbc_build(operator_meta, bundle_build)
+                if existing_fbc_build:
+                    existing_builds.append((dgk, existing_fbc_build))
+
+            if existing_builds:
```

Comment:
If any existing FBC build exists, we still need to rebase and build other non-existing FBCs.

Reply:
ah good catch, updated